FINSApp/src/finsUDP.c

```diff
@@ -553,9 +553,15 @@ int finsUDPInit(const char *portName, const char *address, const char* protocol,
 		}
 	}
 		
-	/* node address is last byte of IP address */
-		
-	pdrvPvt->node = ntohl(pdrvPvt->addr.sin_addr.s_addr) & 0xff;
+	if ( (node != NULL) && (strlen(node) > 0) )
+	{
+		pdrvPvt->node = atoi(node);
```

Comment:
The GUI only allows a number by regular expression, so you would need to have explicitly edited globals.txt or something like that. If you have entered something explicitly it probably means you don't want the default, so both 0 or default last byte are probably equally likely to be wrong?

Reply:
Sure, my worry is that if I accidentally do enter something invalid in the st.cmd (especially as we're making a new st.cmd for each instrument) it won't be very obvious why it doesn't connect but I'm probably being too cautious. Let's leave it as it is.

Reply:
There is no need to change the st.cmd (actually FINS_01.cmd) on any existing instruments, an extra argument has been added and the existing .cmd there will not pass a value which will turn up as NULL above and so maintain the previous behaviour. Only the HLM plc will need to pass a value for this argument and need to have "$(PLCNODE)" added, but it could be added to others if needed.

Reply:
I think it's better to add it anyway to maintain the same behaviour across instruments otherwise if you add a PLC node via the GUI on existing instruments it won't work and it won't be immediately obvious why.

Reply:
After further discussion happy to drop this comment